File: app/src/main/java/com/carbrowser/download/DownloadTask.java
package com.carbrowser.download;

/**
 * Represents a single download task.
 */
public class DownloadTask {

    public static final int STATUS_PENDING = 0;
    public static final int STATUS_RUNNING = 1;
    public static final int STATUS_COMPLETE = 2;
    public static final int STATUS_FAILED = 3;

    public long id;
    public String url;
    public String fileName;
    public String filePath;
    public int status;
    public long totalBytes;
    public long downloadedBytes;
    public int progress;  // 0-100, or -1 if unknown
    public String error;

    public String getStatusText() {
        switch (status) {
            case STATUS_PENDING: return "等待中";
            case STATUS_RUNNING: return "下载中";
            case STATUS_COMPLETE: return "已完成";
            case STATUS_FAILED: return "失败";
            default: return "未知";
        }
    }

    public String getFormattedSize() {
        return formatSize(totalBytes > 0 ? totalBytes : downloadedBytes);
    }

    public String getFormattedDownloaded() {
        return formatSize(downloadedBytes);
    }

    private String formatSize(long bytes) {
        if (bytes <= 0) return "未知";
        if (bytes < 1024) return bytes + " B";
        if (bytes < 1024 * 1024) return String.format("%.1f KB", bytes / 1024.0);
        if (bytes < 1024 * 1024 * 1024) return String.format("%.1f MB", bytes / (1024.0 * 1024));
        return String.format("%.2f GB", bytes / (1024.0 * 1024 * 1024));
    }
}

File: app/src/main/java/com/lazylines/download/DownloadTask.kt
package com.lazylines.download

/**
 * Represents a single download task with its state.
 */
data class DownloadTask(
    val id: Int,
    val url: String,
    val filename: String,
    val savePath: String,
    var totalBytes: Long,
    var downloadedBytes: Long,
    var status: Status
) {
    enum class Status {
        QUEUED,
        DOWNLOADING,
        COMPLETED,
        FAILED,
        CANCELLED
    }

    /** Progress percentage (0-100), returns 0 if total unknown. */
    val progressPercent: Int
        get() = if (totalBytes > 0) ((downloadedBytes * 100) / totalBytes).toInt() else 0
}
